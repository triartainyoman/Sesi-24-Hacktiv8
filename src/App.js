import React, {useState} from 'react';
import {
  View,
  Text,
  TextInput,
  TouchableOpacity,
  StyleSheet,
  Alert,
} from 'react-native';

export default function App() {
  const [name, setName] = useState('');
  return (
    <View style={styles.container}>
      <Text>Absen Nama :</Text>
      <TextInput
        testID="inputNama"
        onChangeText={name => setName(name)}
        style={styles.input}
        placeholder={'Name'}
        value={name}
      />
      <TouchableOpacity
        testID="submitNama"
        onPress={() => Alert.alert('nama', name)}>
        <View style={styles.button}>
          <Text style={styles.textButton}>Absen</Text>
        </View>
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    justifyContent: 'center',
    marginHorizontal: 20,
  },
  input: {
    width: 300,
    height: 40,
    borderWidth: 1,
    borderColor: '#DDD',
    borderRadius: 5,
    marginTop: 10,
  },
  button: {
    width: 300,
    height: 45,
    borderWidth: 1,
    borderColor: '#DDD',
    backgroundColor: '#1B1B32',
    borderRadius: 5,
    marginTop: 10,
    justifyContent: 'center',
    alignItems: 'center',
  },
  textButton: {
    fontWeight: 'bold',
    color: '#FFF',
  },
});
