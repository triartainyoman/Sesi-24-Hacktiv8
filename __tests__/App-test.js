import 'react-native';
import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import {shallow, configure} from 'enzyme';
import renderer from 'react-test-renderer';
import App from '../src/App';
configure({adapter: new Adapter(), disableLifecycleMethods: true});

const profileWrapper = shallow(<App />);

jest.mock('@react-navigation/native', () => ({
  useNavigation: component => component,
}));

describe('App Screen', () => {
  it('should renders correctly', () => {
    renderer.create(<App />);
  });

  it('should renders `App Screen` module correctly', () => {
    expect(profileWrapper).toMatchSnapshot();
  });

  describe('Check component', () => {
    it('should create find `TextInput`', () => {
      expect(profileWrapper.find('TextInput').exists());
    });

    it('should create `TouchableOpacity` component', () => {
      expect(profileWrapper.find('TouchableOpacity').exists());
    });

    describe('Mount component', () => {
      describe('Event Test', () => {
        it('should change text when `Input` Change', () => {
          const mockCallBack = jest.fn();
          profileWrapper.find('[testID="inputNama"]').simulate('change');
          expect(mockCallBack.mock.calls.length);
        });

        it('should click when `Absen` Pressed', () => {
          const mockCallBack = jest.fn();
          profileWrapper.find('[testID="submitNama"]').simulate('press');
          expect(mockCallBack.mock.calls.length);
        });
      });
    });
  });
});
