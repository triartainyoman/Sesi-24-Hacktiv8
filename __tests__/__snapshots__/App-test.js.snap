// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Screen should renders \`App Screen\` module correctly 1`] = `ShallowWrapper {}`;
